### PR TITLE
feat(cdk): add event-manager

### PR DESCRIPTION
**Pull request by BioPhoton**
Not sure if this is worth maintaining or if we can just add it as a snippet. WDYS?

### CODE REVIEW

File: libs/cdk/src/lib/event-manager/shared-event-manager.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+type EventName = string;
+type SharedEvent = Observable<Event> | null | undefined;
+
+interface SharedEventsContainer {
+  window: Map<EventName, SharedEvent>;
+  document: Map<EventName, SharedEvent>;
+}
+
+@Injectable()
+export class ShareEventManager extends EventManager {
```
Comment (Karnaukhov-kh) [inline libs/cdk/src/lib/event-manager/shared-event-manager.ts:18]:
```suggestion
export class SharedEventManager extends EventManager {
```

File: libs/cdk/src/lib/event-manager/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { ShareEventManager } from './shared-event-manager';
```
Comment (Karnaukhov-kh) [inline libs/cdk/src/lib/event-manager/index.ts:1]:
```suggestion
export { SharedEventManager } from './shared-event-manager';
```

**Review by Karnaukhov-kh — COMMENTED** [review]
Can we make it "experimental" and see how it goes? The idea is pretty nice and adds up to the zone-agnostic mental model.

**Comment by BioPhoton** [comment]
I close here and reopen in new PR

**Pull request closed without merging**